packages/cactus-plugin-persistence-fabric/src/main/typescript/plugin-persistence-fabric.ts

```diff
@@ -0,0 +1,914 @@
+    this.log = LoggerProvider.getOrCreate({ level, label });
+    this.instanceId = options.instanceId;
+    this.gatewayOptions = options.gatewayOptions;
+    // this.fabricConnectorPlugin = new PluginLedgerConnectorFabric(options);
+
+    // database
+    this.instanceId = options.instanceId;
+    this.apiClient = options.apiClient;
+
+    this.dbClient = new PostgresDatabaseClient({
+      connectionString: options.connectionString,
+      logLevel: level,
+    });
+  }
+
+  public async shutdown(): Promise<void> {
+    this.apiClient.close();
+    await this.dbClient.shutdown();
+    this.isConnected = false;
+  }
+
+  // public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+  // const { log } = this;
+  // const pkgName = this.getPackageName();
+  // if (this.endpoints) {
+  //   return this.endpoints;
+  // }
+  // this.log.info(`Creating web services for plugin ${pkgName}...`);
+  // const endpoints: IWebServiceEndpoint[] = [];
+  // {
+  //   const endpoint = new StatusEndpointV1({
+  //     connector: this,
+  //     logLevel: this.options.logLevel,
+  //   });
+  //   endpoints.push(endpoint);
+  // }
+  // this.endpoints = endpoints;
+  // this.log.info(`Instantiated web services for plugin ${pkgName} OK`, {
+  //   endpoints,
+  // });
+  // return endpoints;
+  // }
+
+  // public async registerWebServices(
+  //   app: Express,
+  // ): Promise<IWebServiceEndpoint[]> {
+  //   const webServices = await this.getOrCreateWebServices();
+  //   webServices.forEach((ws) => ws.registerExpress(app));
+  //   this.isWebServicesRegistered = true;
+  //   return webServices;
+  // }
+  //
+  //
+  public getStatus(): getStatusReturn {
+    return {
+      instanceId: this.instanceId,
+      connected: this.isConnected,
+      webServicesRegistered: this.isWebServicesRegistered,
+      lastSeenBlock: this.lastSeenBlock,
+    };
+  }
+
+  /**
+   * Get error cause for RuntimeError (instance of `Error`, string or undefined)
+   * @param err unknown error type.
+   * @returns valid `RuntimeError` cause
+   */
+  getRuntimeErrorCause(err: unknown): Error | string | undefined {
+    if (err instanceof Error || typeof err === "string") {
+      return err;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Should be called before using the plugin.
+   * Connects to the database and initializes the plugin schema and status entry.
+   */
+
+  public async onPluginInit(): Promise<void> {
+    await this.dbClient.connect();
+    this.log.info("Connect the PostgreSQL PostgresDatabaseClient");
+    await this.dbClient.initializePlugin(
+      PluginPersistenceFabric.CLASS_NAME,
+      this.instanceId,
+    );
+    this.log.info("Plugin initialized");
+    this.isConnected = true;
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+  // this is just test function to check if you correctly created instance of plugin
+  public async helloWorldTest(): Promise<string> {
```

Comment:
Some developers might need it during progressing with their integration of this plugin within their project.

Reply:
It is used during integrations as one of steps for developers to integrate plugin into their project, without setting up all things around like database or ledger.

Reply:
@barnapa Any chance that you could call it something like healthcheck instead? I want to keep test code and production code separate and with clear lines between the two. What you describe here is a legitimate use-case for production in my opinion but then I'd avoid anything "test" in its name for that exact reason just so that it doesn't confuse people.